drivers/watchdog/wdt_rpi_pico.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2022, Jamie McCrae
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_pico_watchdog
+
+#include <hardware/watchdog.h>
+#include <hardware/structs/psm.h>
+#include <drivers/watchdog.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wdt_rpi_pico, CONFIG_WDT_LOG_LEVEL);
+
+/* Maximum watchdog time is halved due to errata RP2040-E1 */
+#define RPI_2040_MAX_WDT_TIME (0xffffff / 2)
```

Comment:
`RPI_PICO` would be a better prefix, since this driver is not just for RP2040. But anyway, I don't think you even need to namespace it, since the definition is not exported outside of this file.

Reply:
Updated to RPI_PICO